pypesto/objective/aggregate.py

```diff
@@ -0,0 +1,175 @@
+import numpy as np
+
+from .objective import Objective
+
+
+class AggregateObjective(Objective):
+    """
+    This class allows to create an aggregateObjective from a list of
+    Objective instances.
+    """
+
+    def __init__(self, objectives, x_names=None, options=None):
+        """
+        Constructor.
+
+        Parameters
+        ----------
+
+        objectives: list
+            List of pypesto.objetive instances
+
+        """
+        # input typechecks
+        if not isinstance(objectives, list):
+            raise TypeError(f'Objectives must be a list, '
+                            f'was {type(objectives)}.')
+
+        if not all(
+                isinstance(objective, Objective)
+                for objective in objectives
+        ):
+            raise TypeError('Objectives must only contain elements of type'
+                            'pypesto.Objective')
+
+        self.objectives = objectives
+
+        # assemble a dict that we can pass as kwargs to the
+        # pypesto.Objective constructor
+        init_kwargs = {
+            'x_names': x_names,
+            'options': options
+        }
+
+        # check if all objectives consistently accept sensi orders in fun/res
+        # and adopt the same behaviour in aggregate
+        for attr in ['fun_accept_sensi_orders', 'res_accept_sensi_orders']:
+            _check_boolean_value_consistent(objectives, attr)
+            init_kwargs[attr] = getattr(objectives[0], attr)
+
+        # check if all objectives consistently accept sensi orders in fun/res
+        # and adopt the same behaviour in aggregate
+        for attr in [
+            'fun', 'grad', 'hess', 'hessp', 'res', 'sres'
+        ]:
+            if any(
+                getattr(objective, attr) is None
+                for objective in objectives
+            ):
```

Comment:
if i understand correctly, attr is set to None as soon as any objective.attr is None? Maybe we want to check that all are None, to have consistency, just as done for boolean values?